http.go

```diff
@@ -302,6 +265,54 @@ func postOutboundMessageHandler(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintf(w, "Message posted (%.2f kB)", float64(buf.Len()/1024))
 }
 
+func attachFile(w http.ResponseWriter, f *multipart.FileHeader, msg *fbb.Message) error {
```

Comment:
I really like that you moved this out to separate function. It really cleans up the handler 🙂 

I think it would be even better if we could remove the ResponseWriter argument, to make it both less verbose and get rid of the HTTP response details in this function.

Of course, we still need to support returning different status codes based on the nature of the errors. One approach I've seen (and used) multiple times, is by defining a custom error type that wraps the original error and adds a StatusCode field.

https://play.golang.org/p/FxGw97vp2Gr

How do you feel about that?

Reply:
Done!